### ISSUE TITLE
meteor npm dependencies

### ISSUE THREAD

**Issue by joshua1** [Urigo/meteor-client-bundler#18]
I am using aldeed's Collection2 hich has a dependency on Simpl-Schema a node package, since the bundler does not make reference to node modules, i get an error stating that 
``` 
Uncaught Error: Cannot find module 'simpl-schema'
```
I have added this package to the client application but still get the same error. what is the best way to deal with this?

**Comment by pal-pinter** [comment]
May you have wrong type.
'simpl-schema' -> 'simple-schema'

**Comment by joshua1** [comment]
No it's not the wrong type.the package is name simpl-schema on npm. I have taken it out though so that I could move on 

joshua O

On Fri, Apr 28, 2017 at 9:38 AM "Pintér Pál"

<
mailto:
> wrote:

a, pre, code, a:link, body { word-wrap: break-word !important; }

May you have wrong type.

'simpl-schema' -> 'simple-schema'

—

You are receiving this because you authored the thread.

Reply to this email directly,
https://github.com/Urigo/meteor-client-bundler/issues/18#issuecomment-297930091
, or

**Comment by KeithGillette** [comment]
I am experiencing a similar problem with the `meteor-apollo-integration` Atmosphere package, which has a dependency on `apollo-client`. I get a browser console error `Uncaught Error: Cannot find module 'apollo-client'` when including `apollo` in my `meteor-client.config.json` even though `apollo-client` is already installed in `node_modules` as a project dependency in `package.json`.

**Comment by trymbill** [comment]
Same here, trying to use Collections2 and I get an import error ```Uncaught Error: Cannot find module 'simpl-schema'```. I tried adding it to the import section of my bundle-config.json but that didn't fix it.

**Comment by KeithGillette** [comment]
Any recommended work-arounds or plans to address this issue, @DAB0mB or @Urigo? I fear that lack of support for using bundled Atmosphere packages that also have dependencies on npm modules may prevent me from using Angular-Meteor in my project. Thanks for any thoughts.

**Comment by untsamphan** [comment]
I faced the same issue because my project use `std:accounts-ui` which require `react` from npm. When I removed it (use AccountsUIWrapper) the problem gone.

I guess Meteor packages that require any npm module' will fail when bundled.

**Comment by briancvn** [comment]
I have same error aldeed:meteor-collection2-core
![error](https://user-images.githubusercontent.com/11416798/28903144-800374ce-782d-11e7-81db-0bb567bc2485.png)

**Comment by RDeluxe** [comment]
Same issue here. I have to say, all this setup and research to put in place a basic app is getting on my nerves 😭

**Comment by whtsky** [comment]
#48 
for now, you can `npm i -g @whtsky/meteor-client-bundler` and bundle with config like
```json
{
  "release": "1.5.2",
  "runtime": {},
  "import": [
    "meteor-base@1.1.0",
    "mongo@1.1.19",
    "tracker@1.1.3",
    "accounts-google@1.2.0",
    "accounts-base@1.3.1",
    "aldeed:collection2-core@2.0.0",
  ],
  "npmPackages": ["simpl-schema@^0.3.2"]
}

```

**Comment by psnfrench** [comment]
Is there any update as to how to get this working? I have tried installing the @whtsky version as above, but it gives me `npm ERR! No valid versions available for @whtsky/meteor-client-bundler`

**Comment by whtsky** [comment]
@psnfrench I might have unpublished it by mistake, sorry.
should works now

**Comment by psnfrench** [comment]
Thanks @whtsky it works perfectly now. I hope this get merged into the master branch soon as its quite limiting without it.

**Comment by Urigo** [comment]
merged and released in `0.5.0`.   took a while but finally landed, sorry about that

**Comment by gustawdaniel** [comment]
Now you can fix this error adding `simpl-schema` to list of packages in file `meteor-client.config.json`

```
{
  "npmPackages": [ "simpl-schema@^1.5.3" ]
}
```

I confirm it works.
